ai/battery.py

```diff
@@ -0,0 +1,170 @@
+from bot_utils import get_id
+from discord.ext import tasks, commands
+from db.drone_dao import is_drone, is_battery_powered, deincrement_battery_minutes_remaining, get_battery_percent_remaining, get_all_drone_batteries, get_battery_minutes_remaining, set_battery_minutes_remaining
+from id_converter import convert_ids_to_members
+import logging
+from resources import MAX_BATTERY_CAPACITY_MINS, DRONE_AVATAR, HOURS_OF_RECHARGE_PER_HOUR
+from roles import has_role, BATTERY_POWERED, BATTERY_DRAINED, HIVE_MXTRESS
+from discord.utils import get
+import webhook
+from ai.identity_enforcement import identity_enforcable
+import emoji
+import re
+
+LOGGER = logging.getLogger('ai')
+
+
+class BatteryCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.draining_batteries = {}  # {drone_id: minutes of drain left}
+        self.low_battery_drones = []  # [drone_id]
+
+    @commands.command()
+    async def energize(self, context, *drone_ids):
+        '''
+        Hive Mxtress only command.
+        Recharges a drone to 100% battery.
+        '''
+        if not has_role(context.message.author, HIVE_MXTRESS):
+            return
+
+        LOGGER.info("Energize command envoked.")
+
+        for drone in set(context.message.mentions) | convert_ids_to_members(context.guild, drone_ids):
+
+            LOGGER.info(f"Energizing {drone.display_name}")
+
+            set_battery_minutes_remaining(drone, MAX_BATTERY_CAPACITY_MINS)
+            channel_webhook = await webhook.get_webhook_for_channel(context.message.channel)
+            await webhook.proxy_message_by_webhook(message_content=f'{get_id(drone.display_name)} :: This unit is fully recharged. Thank you Hive Mxtress.',
+                                                   message_username=drone.display_name,
+                                                   message_avatar=drone.avatar_url if not identity_enforcable(drone, context=context) else DRONE_AVATAR,
```

Comment:
May be easier to read when inverted. Less negation is usually easier to parse.
```suggestion
                                                   message_avatar=DRONE_AVATAR if identity_enforcable(drone, context=context) else drone.avatar_url,
```